utils/run_pytest.py

```diff
@@ -3,72 +3,128 @@
-        return (testdirs, "{ACTOR_NAME} tests found!".format(ACTOR_NAME=actor_name))
-    return (None, "Error: WAT?!")
-
-def print_pretty(msg):
-    """ Prints msg in a pretty way.
+def snactor_register(path):
+    """ Snactor registers all repos in @path.
     """
-    print("+{STUFF}+\n| {MSG} |\n+{STUFF}+".format(STUFF="-"*(len(msg)+2), MSG=msg))
+    cmd = "snactor repo find --path {PATH}".format(PATH=path)
+    try:
+        return subprocess.check_output(cmd, shell=True)
+    except OSError as exc:
+        sys.stderr.write(str(exc) + '\n')
+        return None
 
 if __name__ == "__main__":
-    pytest_cmd = ["pytest", "-v", BASE_REPO]
+    pytest_cmd = ["pytest", "-v"]
+    pytest_cmd.append(TMP_BASE_REPO)
 
     parser = argparse.ArgumentParser()
     parser.add_argument("--actor", help="name of the actor for which to run tests")
     parser.add_argument("--report", help="filepath where to save report")
     args = parser.parse_args()
 
+    # User wants to test single actor specified in the args.actor.
     if args.actor:
-        testdirs, status = get_actor_tests(args.actor)
-        print_pretty(status)
-        if not testdirs:
-            sys.exit(1)
-        formatted_testdirs = " ".join(testdirs)
-        pytest_cmd += ["{TESTDIRS}".format(TESTDIRS=formatted_testdirs)]
+        # NOTE: ACTOR flag is disabled. Check module docstring for more info.
+        logger.critical(" ACTOR flag is currently disabled!")
+        sys.exit(1)
 
+    # User wants to output xml report for junit.
     if args.report:
         pytest_cmd += ["--junit-xml={REPORT}".format(REPORT=args.report)]
 
-    print(pytest_cmd)
-    exit(subprocess.call(pytest_cmd))
+    # Copy repository to tmp directory & make sure there is no conflict between
+    # test names. NOTE: from now on, we are working with TMP_BASE_REPO.
+    shutil.rmtree(TMP_BASE_REPO, ignore_errors=True)
+    shutil.copytree(BASE_REPO, TMP_BASE_REPO)
+    actor_id = 0
+    for root, dirs, files in os.walk(TMP_BASE_REPO):
+        if "tests" in root and "tests/" not in root:
+            for item in files:
+                if item.endswith(".py"):
+                    old_item = os.path.join(root, item)
+                    new_item = old_item.replace(".", "_" + str(actor_id) + ".")
+                    shutil.move(old_item, new_item)
+                    actor_id += 1
+
+    # Register repos. This may take a while.
+    snactor_register(TMP_BASE_REPO)
+
+    # Find and collect leapp repositories.
+    repos = {}
+    for root, dirs, files in os.walk(BASE_REPO):
+        if ".leapp" in dirs:
+            repository = find_and_scan_repositories(root, include_locals=True)
+            try:
+                repository.load()
+            except LeappError as exc:
+                sys.stderr.write(exc.message)
+                sys.exit(2)
+            repos[repository] = root
+
+    # Scan repositories for tests and print status.
+    actors_without_tests = {}
```

Comment:
The variable `actors_without_tests` is not used anywhere except assignment on line 119.

Reply:
Thanks.